### ISSUE TITLE
Java Tensorflow save & restore without using filesystem

### ISSUE THREAD

**Issue by cosmir17** [tensorflow/tensorflow#26133]
**System information**
- TensorFlow version (you are using): Up to date
- Are you willing to contribute it (Yes/No): 

**Describe the feature and the current behaviour/state.**
I previously opened a feature ticket and asking for a **python** approach.
https://github.com/tensorflow/tensorflow/issues/25839

I wasn't aware there is **java tensorflow**.

I would like to retrieve(instead of Save) training information and restore it to a Tensorflow session when necessary (without using a file system). I want to bypass filesystem.

### I want to know a line of code or method that I can use to extract training data(as an object) and import it to (tf)session when necessary without me handling file path etc

to elaborate =>
I found that there is a way to retrieve updated weight values.
https://www.quora.com/How-do-I-print-weights-of-a-fully-connected-neural-network-in-tensorflow
//train your network here
sess.run(optimizer, feed_dict={x:data, y:labels}
print("layer 1 weights:", sess.run(W1))
print("layer 2 weights:", sess.run(W2))

**Rather than handling each individual tensor information, I want to extract the whole training information (as an object) and import it to (tf)session when necessary.**

I have looked at the followings and searched but was not successful.
https://www.tensorflow.org/guide/saved_model
https://stackoverflow.com/questions/33759623/tensorflow-how-to-save-restore-a-model

Why does this feature not appear already present?

**Will this change the current API? How?**
no, I just need(or to know) an additional API.

**Who will benefit with this feature?**
It will make the use of Tensorflow lib much more flexible.
**Any Other info.**

**Comment by cosmir17** [comment]
@allenlavoie @jvishnuvardhan   Thank you for working on this feature! I really appreciate it 👍

**Comment by tilakrayal** [comment]
@cosmir17,
Have you got the chance to check **SavedModelBundle** and **SavedModelBundle.Loader**, where SavedModelBundle represents a model loaded from storage and SavedModelBundle.Loader will provide the options for loading a SavedModel.



SavedModelBundle | load()Load a SavedModelBundle with the configured options.
-- | --
SavedModelBundle.Loader | withConfigProto(byte[] configProto)Set configuration of the Session object created when loading the model.
SavedModelBundle.Loader | withRunOptions(byte[] options)Sets options to use when executing model initialization operations.
SavedModelBundle.Loader | withTags(String... tags)Sets the set of tags that identify the specific graph in the saved model to load.

https://www.tensorflow.org/api_docs/java/org/tensorflow/SavedModelBundle
https://www.tensorflow.org/api_docs/java/org/tensorflow/SavedModelBundle.Loader

Thank you!
